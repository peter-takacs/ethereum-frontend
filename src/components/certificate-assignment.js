import React from 'react';

const CertificateAdder = ({status, onClick, candidate, certificate, onChange}) => {
    let addressInput;
    let certificateInput;

    let handleChange = () => {
        onChange(addressInput.value, certificateInput.value);
    }

    return (
        <div>
            <header>
            Certificate adder 
            </header>
            <div>
            <input type="text" id="address" placeholder="Candidate address" value={candidate} ref={node => addressInput = node} onChange={handleChange}/>
            <input type="text" id="certificate" placeholder="Certificate to add" value={certificate} ref={node => certificateInput = node} onChange={handleChange}/>
            <button onClick={() => onClick(candidate, certificate)}>Submit</button>
            </div>
        </div>
    );
}

export default CertificateAdder;