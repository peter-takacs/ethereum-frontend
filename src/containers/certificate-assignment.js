import { connect } from 'react-redux';
import CertificateAdder from '../components/certificate-assignment';
import { requestAddition, changeQuery } from '../actions/certificate-assignment-actions';

const mapDispatchToProps = dispatch => ({
    onClick: (candidate, certificate) => dispatch(requestAddition(candidate, certificate)),
    onChange: (candidate, certificate) => dispatch(changeQuery(candidate, certificate))
})

const mapStateToProps = (state) => {
    return {
        candidate: state.certificateAssignment.candidate,
        certificate: state.certificateAssignment.certificate,
        status: state.certificateAssignment.status
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(CertificateAdder);