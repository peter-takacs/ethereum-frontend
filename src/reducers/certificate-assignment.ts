
import {
    REQUEST_SENT,
    CHANGE_ASSIGNMENT_QUERY,
    Actions
} from '../actions/certificate-assignment-actions';

const certificateAssignment = (state = {candidate: '', status: '', certificate: ''}, action: Actions) => {
    switch (action.type) {
        case CHANGE_ASSIGNMENT_QUERY:
            return {
                ...state,
                candidate: action.query.candidate || state.candidate,
                certificate: action.query.certificate || state.certificate
            }
        case REQUEST_SENT:
            return {
                ...state,
                status: 'DONE'
            }
        default:
            return state;
    }
}

export default certificateAssignment;