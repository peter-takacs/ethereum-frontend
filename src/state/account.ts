import { Address } from "../types/ethereum-address";

export interface AccountState {
    address: Address | null;
}