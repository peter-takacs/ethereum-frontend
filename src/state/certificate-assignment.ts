export enum CertificateQueryState {
    Idle,
    Loading,
    Assigned,
    NotAssigned
}

export interface State {
    candidate: string,
    certificate: string,
    state: CertificateQueryState
}