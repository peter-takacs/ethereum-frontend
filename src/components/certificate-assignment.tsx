import * as React from 'react';
import { State } from '../state/certificate-assignment';

export interface CertificateAdderProps extends State {
    onClick: (candidate: string, certificate: string) => void;
    onChange: (candidate: string, certificate: string) => void;
}

const CertificateAdder = ({state, onClick, candidate, certificate, onChange}: CertificateAdderProps) => {
    let addressInput: HTMLInputElement;
    let certificateInput: HTMLTextAreaElement;

    let handleChange = () => {
        onChange(addressInput.value, certificateInput.value);
    }

    return (
        <div>
            <header>
            Certificate adder 
            </header>
            <div>
            <input type="text" id="address" placeholder="Candidate address" value={candidate} ref={node => addressInput = node} onChange={handleChange}/>
            <textarea id="certificate" placeholder="Certificate to add" value={certificate} ref={node => certificateInput = node} onChange={handleChange}/>
            <button onClick={() => onClick(candidate, certificate)}>Submit</button>
            </div>
        </div>
    );
}

export default CertificateAdder;