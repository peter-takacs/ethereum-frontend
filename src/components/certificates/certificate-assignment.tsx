import * as React from 'react';
import { CertificateAdderState } from '../../state/certificate-assignment';
import { FormControl, Input, Button, TextField, Grid } from '@material-ui/core';
import { sha256 } from 'js-sha256';
import { Address } from '../../types/ethereum-address';
import AddressEditor from '../common/address-editor';


export interface CertificateAssignmentDispatch {
    onClick: (candidate: Address, certificate: string) => void;
}

export type CertificateAdderProps = CertificateAdderState & CertificateAssignmentDispatch;

interface State {
    candidate: Address | null;
    certificate: string;
}

export class CertificateAdder extends React.Component<CertificateAdderProps, State> {
    
    private addressChange(candidate: Address) {
        this.setState({
            candidate 
        })
    }
    private certificateChange(certificate: string) {
        this.setState({
            certificate 
        })
    }

    private isValid(): boolean {
        if (this.state.candidate == null || this.state.certificate == null) {
            return false;
        }
        return true;
    }

    private onSubmitClicked() {
        if (this.isValid()) {
            this.props.onClick(this.state.candidate!, this.state.certificate);
        }
    }

    constructor(props: CertificateAdderProps) {
        super(props);
        this.state = {
            candidate: null,
            certificate: ''
        }
    }

    public render() {
        return (
            <Grid container spacing={24}>
                <Grid item xs>
                    <FormControl>
                        <AddressEditor
                            address={this.state.candidate}
                            placeholder="Candidate address"
                            onChange={(address) => address && this.addressChange(address)}
                        />
                        <TextField type="text" id="certificate"
                            placeholder="Certificate to add"
                            value={this.state.certificate}
                            onChange={(certificate) => this.certificateChange(certificate.target.value)}
                            multiline={true}
                            rows={4} />
                        <Button 
                            onClick={() => this.onSubmitClicked()}
                            disabled={!this.isValid()}
                        >
                            Submit
                        </Button>
                    </FormControl>
                </Grid>
                <Grid item xs>
                    Hash of current assertion to be submitted: {sha256(this.state.certificate)}
                </Grid>
            </Grid>
        );
    }

}

export default CertificateAdder;