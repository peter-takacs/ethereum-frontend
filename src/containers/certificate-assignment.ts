import CertificateAdder, { CertificateAssignmentDispatch } from '../components/certificates/certificate-assignment';
import { requestAddition } from '../actions/certificate-assignment-actions';
import { connect } from 'react-redux';
import { State as RootState } from '../state/root';
import { CertificateAdderState as CertificateAssignmentState} from '../state/certificate-assignment';
import { ThunkDispatch } from 'redux-thunk';
import { Actions } from '../actions/certificate-assignment-actions';
import { Address } from '../types/ethereum-address';

const mapDispatchToProps = (dispatch: ThunkDispatch<CertificateAssignmentState, undefined, Actions>) => ({
    onClick: (candidate: Address, certificate: string) => dispatch(requestAddition(candidate, certificate)),
})

const mapStateToProps = (state: RootState) => {
    return {
        state: state.certificateAssignment.state
    }
}
export default connect<CertificateAssignmentState, CertificateAssignmentDispatch, {}, RootState>(
    mapStateToProps, 
    mapDispatchToProps)
(CertificateAdder);