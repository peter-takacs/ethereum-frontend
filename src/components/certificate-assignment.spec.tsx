import * as React from 'react';
import * as renderer from 'react-test-renderer';
import CertificateAdder from './certificate-assignment';
import { CertificateQueryState } from '../state/certificate-assignment';
import { dummyAddress } from '../utils/test/address';

test('Renders correctly in every state', () => {
    [
        CertificateQueryState.Assigned,
        CertificateQueryState.Idle,
        CertificateQueryState.Loading,
        CertificateQueryState.NotAssigned
    ].forEach(state => {
        const component = renderer.create(
            <CertificateAdder
                candidate={null}
                certificate="0x1234"
                onCandidateChange={() => { }}
                onCertificateChange={() => { }}
                onClick={() => { }}
                state={state} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

test('Renders correctly with a correct address', () => {
    const component = renderer.create(
        <CertificateAdder
            candidate={dummyAddress(0)}
            certificate="dummy"
            onCandidateChange={() => {}}
            onCertificateChange={() => {}}
            onClick={() => {}}
            state={CertificateQueryState.Idle}
        />);
    expect(component.toJSON()).toMatchSnapshot();
});


test('Renders correctly with an empty address', () => {
    const component = renderer.create(
        <CertificateAdder
            candidate={null}
            certificate="dummy"
            onCandidateChange={() => {}}
            onCertificateChange={() => {}}
            onClick={() => {}}
            state={CertificateQueryState.Idle}
        />);
    expect(component.toJSON()).toMatchSnapshot();
});