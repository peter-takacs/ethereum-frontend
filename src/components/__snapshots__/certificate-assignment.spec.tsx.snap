// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly in every state 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-multiline-108 MuiInput-underline-106"
        >
          <textarea
            aria-invalid={false}
            className="MuiInput-input-110 MuiInput-inputMultiline-112"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to add"
            required={false}
            rows={4}
            type="text"
            value="0x1234"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Hash of current assertion to be submitted: 
    5a0737e8cbcfa24dcc118b0ab1e6d98bee17c57daa8a1686024159aae707ed6f
  </div>
</div>
`;

exports[`Renders correctly in every state 2`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-multiline-108 MuiInput-underline-106"
        >
          <textarea
            aria-invalid={false}
            className="MuiInput-input-110 MuiInput-inputMultiline-112"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to add"
            required={false}
            rows={4}
            type="text"
            value="0x1234"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Hash of current assertion to be submitted: 
    5a0737e8cbcfa24dcc118b0ab1e6d98bee17c57daa8a1686024159aae707ed6f
  </div>
</div>
`;

exports[`Renders correctly in every state 3`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-multiline-108 MuiInput-underline-106"
        >
          <textarea
            aria-invalid={false}
            className="MuiInput-input-110 MuiInput-inputMultiline-112"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to add"
            required={false}
            rows={4}
            type="text"
            value="0x1234"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Hash of current assertion to be submitted: 
    5a0737e8cbcfa24dcc118b0ab1e6d98bee17c57daa8a1686024159aae707ed6f
  </div>
</div>
`;

exports[`Renders correctly in every state 4`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-multiline-108 MuiInput-underline-106"
        >
          <textarea
            aria-invalid={false}
            className="MuiInput-input-110 MuiInput-inputMultiline-112"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to add"
            required={false}
            rows={4}
            type="text"
            value="0x1234"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Hash of current assertion to be submitted: 
    5a0737e8cbcfa24dcc118b0ab1e6d98bee17c57daa8a1686024159aae707ed6f
  </div>
</div>
`;

exports[`Renders correctly with a correct address 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value="0xdf3f619804a92fdb4057192dc43dd748ea778adc"
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-multiline-108 MuiInput-underline-106"
        >
          <textarea
            aria-invalid={false}
            className="MuiInput-input-110 MuiInput-inputMultiline-112"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to add"
            required={false}
            rows={4}
            type="text"
            value="dummy"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Hash of current assertion to be submitted: 
    b5a2c96250612366ea272ffac6d9744aaf4b45aacd96aa7cfcb931ee3b558259
  </div>
</div>
`;

exports[`Renders correctly with an empty address 1`] = `
<div
  className="MuiGrid-container-1 MuiGrid-spacing-xs-24-25"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    <div
      className="MuiFormControl-root-98"
    >
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-underline-106"
        >
          <input
            aria-invalid={false}
            className="MuiInput-input-110"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Candidate address"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="MuiFormControl-root-98"
      >
        <div
          className="MuiInput-root-102 MuiInput-formControl-103 MuiInput-multiline-108 MuiInput-underline-106"
        >
          <textarea
            aria-invalid={false}
            className="MuiInput-input-110 MuiInput-inputMultiline-112"
            disabled={false}
            id="certificate"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Certificate to add"
            required={false}
            rows={4}
            type="text"
            value="dummy"
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root-141 MuiButtonBase-disabled-142 MuiButton-root-115 MuiButton-text-117 MuiButton-flat-120 MuiButton-disabled-135"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="button"
      >
        <span
          className="MuiButton-label-116"
        >
          Submit
        </span>
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-true-29"
  >
    Hash of current assertion to be submitted: 
    b5a2c96250612366ea272ffac6d9744aaf4b45aacd96aa7cfcb931ee3b558259
  </div>
</div>
`;
